domain/model/service/modelservice.go

```diff
@@ -46,23 +49,29 @@ type ControllerState interface {
 // ModelService defines a service for interacting with the underlying model
 // state, as opposed to the controller state.
 type ModelService struct {
-	clock        clock.Clock
-	modelID      coremodel.UUID
-	controllerSt ControllerState
-	modelSt      ModelState
+	clock                 clock.Clock
+	modelID               coremodel.UUID
+	controllerSt          ControllerState
+	modelSt               ModelState
+	environProviderGetter EnvironVersionProviderFunc
 }
 
+// TODO(alvin): we need to extract model creation/deletion into a separate service so that this
+// service operates only on the current model and we no longer need to pass in the UUID when
+// creating the service.
```

Comment:
Sorry @CodingCookieRookie I know @wallyworld requested this change but I would like you to undo it. It is not the job of this PR to document design decision that we are not happy with anymore. We can safely assume this get's done because there is a task setup to track this work in our backlog.

On top of this TODO's are where things go to die in the code base. We will get this work done and if something needs to be added it should be done in a follow up PR and not side loaded into this one.